File: Algorithms/C/62 Unique Paths.c
int uniquePaths(int m, int n){
    int* dp = (int*)malloc(sizeof(int)*n);
    for (int i = 0; i < n; ++i){
        dp[i] = 1;
    }
    for (int i = 1; i < m; ++i)
        for (int j = 0; j < n; ++j){
            if (j == 0){
                dp[j] = 1;
            }
            else{
                dp[j] = dp[j] + dp[j - 1];
            }
        }
    int result = dp[n - 1];
    free(dp);
    return result;
}

File: Algorithms/Python/62 Unique Paths.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        dp = [1] * n
        for i in range(1, m):
            for j in range(n):
                if j == 0:
                    dp[j] = 1
                else:
                    dp[j] = dp[j - 1] + dp[j]
        return dp[n - 1]
